### ISSUE TITLE
targetElement should be optional

### ISSUE THREAD

**Issue by fzembow** [willmcpo/body-scroll-lock#55]
I don't agree with the current behavior (at least in iOS simulator) where `disableBodyScroll` does not work if you don't provide a targetElement 

It should either:

1. Error when a targetElement is not provided, warning the developer that one is required 
2. (_better_): Allow a null targetElement, wherein everything is prevented from scrolling.

**Comment by Jokero** [comment]
Totally agree. Having mandatory targetElement looks very unintuitive

**Comment by willmcpo** [comment]
A target element needs to be provided particularly for ios - the purpose of this library is to lock the background whilst maintaining scrollability of a `targetElement`.

I will instead add a warning

**Comment by fzembow** [comment]
> so nothing on the screen is scrollable?

Yeah. I really appreciate that targetElement exists, for things like side navs whose content might overflow, but I had a use case where I was just overlaying a modal with a single, centered piece of text, and I specifically did _not_ want it to be scrollable.

> If you don't mind, could you submit a PR?

Sure, I'll put together a PR this evening :)

**Comment by willmcpo** [comment]
Great - that'd be awesome, thank you.

**Comment by willmcpo** [comment]
Although the PR is merged, im not convinced this should be default behaviour 

If you try the demo with an Android, or non iOS device, and disable without target element, the target is still scrollable, but isn't in iOS.

Great for iOS, but the inconsistent behaviour is bothering me.

**Comment by fzembow** [comment]
Thanks for taking a look at the PR!

Do you suggest that we disable _all_ scrollable elements without a `targetElement`?

IMO that always felt like a workaround on iOS, rather than a feature of that platform's implementation. But I agree with you that cross-platform consistency is somewhat lacking.

**Comment by willmcpo** [comment]
Hey sorry I think I will revert this PR or we'll need to iterate on master. I think the better approach is to enforce a target element is provided (or throw error if it isn't).

If you then want target to also be locked, you pass an option `lockTarget: boolean` with false being default.

For iOS, we continue doing what you did if locktarget: true

For Android and other platforms, we disable scrolling on body and target if lockTarget: true.

The goal of this library was always to lock the body, nothing else. So this scenario is the default. Any edge cases that want to do things like also locking target should be opt in.

Sorry I should've taken more time with this.

I'll revert the PR for now - if you want the, you could implement the above, otherwise I'll do it when I find time next!
